docs/src/gallery/tutorials/run_annoy.py

```diff
@@ -126,19 +128,16 @@
 import numpy as np
 
 def avg_query_time(annoy_index=None, queries=1000):
-    """
-    Average query time of a most_similar method over 1000 random queries,
-    uses annoy if given an indexer
-    """
+    """Average query time of a most_similar method over 1000 random queries."""
     total_time = 0
     for _ in range(queries):
-        rand_vec = model.wv.vectors_norm[np.random.randint(0, len(model.wv.vocab))]
+        rand_vec = model.wv.vectors_norm[np.random.randint(0, len(model.wv))]
```

Comment:
Since `vectors_norm` is no longer a persistent cache of all unit-normed vectors, but rather (for backward-compatibility) re-calculated every time it's requested, this approach will now involve a lot of redundant recalculation of the whole array. Options: (a) grab it once before the loop; (b) adapt to use `get_vector(random_index, use_norm=True)`.

Reply:
Oh that's sneaky. Isn't it better to fail outright, rather than do an expensive operation silently, where existing code expects a cheap dict lookup?

Reply:
Or, we want to keep backward comp, norm just one vector… I haven't checked the code but why would we recalc the whole array for a single `__getitem__`?

Reply:
We're *not* recalculating for any specific request of a single vector - that does the minimum, and would be a quickie option ((b) above) to improve this code. 

But if a caller asks for `vectors_norm`. it's now calculated & returned (without being cached). Arguably, `vectors_norm` was barely even part of the public API, as it wasn't guaranteed to even exist if a caller hadn't triggered `init_sims()` yet. I woudn't expect the problematic pattern here, requesting `vectors_norm` repeatedly, to be very common, especially when callers who just need single vectors have the `get_vector(..., use_norm=True)` option. 

I suppose we could (a) cache it once requested, to be more like the old behavior, but then we have to add more logic for properly handling other invalidations of this cache elsewhere, and explicitly discarding it at times (like serialization); or (b) eliminate the backward-compatible `@getter` property, either leaving it blank (to generate cryptic error), or stub an explanatory error message. In that latter case, a new distinctly-name utility method could be added, `calculate_unit_normed_vectors()`, that avoids creating the impression this is a simple access of an already-existing property.

Reply:
I'm in favour or removing it = hard fail with a clear message (again, noob-friendly = prescription for "what to do next").

-1 on pretending we're a simple dict, when there's a substantial calculation happening behind the scenes.

Reply:
Sure, `.vectors_norm` should be a hard-fail with helpful error message, similar to the one for `init_sims()`. I can add these for both as part of other touch-up work, such as that in #2892.

Reply:
Alright. I'll add some tentative fix here (so I don't merge "known to be bad" code), merge this PR, and then let's continue with this in another PR, such as #2892.

Reply:
@gojomo pushed now. Let me know if OK and I'll merge this PR, to be continued elsewhere.